### ISSUE TITLE
Non-fluent flushers

### ISSUE THREAD

**Issue by M-Pixel** [M-Pixel/FastestBinaryStream#2]
In #1, @silkfire mentioned,

> The fluent design isn't always of convenience, sometimes you just want to return the data you've read instead of the BinaryStream object.

Indeed, non-fluent value-returning overloads for all Flush methods are necessary to simplify this...

```csharp
byte[] SomePureMethod(SomethingWorthSerializing x)
{
    new BinaryStream()
        .Foo(...)
        .Bar(...)
        .Baz(...)
        .FlushByteArray(out var bytes)
        .Dispose();
    return bytes;
}
```

... into this...

```csharp
byte[] SomePureMethod(SomethingWorthSerializing x) => new BinaryStream()
    .Foo(...)
    .Bar(...)
    .Baz(...)
    .FlushByteArrayAndDispose();
```

**Comment by M-Pixel** [comment]
Fixed by a3078cf

**Comment by M-Pixel** [comment]
... and by 7ae4caee
